core/capabilities/ccip/common/extradatacodec.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import (
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+type ExtraDataCodec struct{}
+
+func NewExtraDataCodec() ExtraDataCodec {
+	return ExtraDataCodec{}
+}
+
+func (c ExtraDataCodec) DecodeExtraArgs(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
```

Comment:
Looks like you're missing `DecodeExtraData`, I thought you were just adding `DecodeTokenAmountDestExecData`. You'll need to add the old value and deprecate/remove it in a future PR.

Reply:
I was renaming it from `DecodeExtraData ` to `DecodeExtraArgs ` since it wasn't accurate.